sdk/core/core-http/src/serviceClient.ts

```diff
@@ -996,7 +996,9 @@ export function flattenResponse(
   } => {
     return Object.defineProperty(obj, "_response", {
       value: _response,
-    });
+    }) as {
```

Comment:
shouldn't this intersect with the passed in type? like if we made this a generic method that took a `T extends Record<string, unknown>` this would be `T & { _response: HttpOperationResponse }`